UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs

```diff
@@ -155,6 +165,19 @@ private void UndertaleRoomEditor_DataContextChanged(object sender, DependencyPro
 
                 if (room.Layers.Count > 0) // if GMS 2+
                 {
+                    LayerZIndexConverter.ProcessOnce = true;
+
+                    bool ordered = true;
+                    for (int i = 0; i < room.Layers.Count - 1; i++)
+                    {
+                        if (room.Layers[i].LayerDepth > room.Layers[i + 1].LayerDepth)
+                            ordered = false;
+                    }
+
+                    if (!ordered)
+                        if (MainWindow.ShowQuestion("Room layers are not ordered by depth.\nRearrange them?", MessageBoxImage.Warning) == MessageBoxResult.Yes)
+                            room.RearrangeLayers();
```

Comment:
```suggestion
                    if (!ordered)
                    {
                        if (MainWindow.ShowQuestion("Room layers are not ordered by depth.\nRearrange them?", MessageBoxImage.Warning) == MessageBoxResult.Yes)
                            room.RearrangeLayers();
                     }
```

Reply:
Also, not sure yet if popping up this question every time could get annoying.

Reply:
I've thought about it.
What if ask another question on _"No"_, like _"Don't ask this question for this session?"_ ?

Reply:
Personally, not that much of a fan of that.

Reply:
So, then I guess it should stay as is _(curly brackets too)_.

Reply:
Yeah, if it gets annoying at one point, someone'll probably complain. Or me.